src/pyodide-js/src/pyodide.ts

```diff
@@ -0,0 +1,292 @@
+              const packageUri = toLoad[pkg];
+              if (packageUri != 'default channel') {
+                return packageUri.replace(/\.js$/, ".data");
+              };
+            };
+            return this.baseUrl + path;
+        };
+
+
+        const promise: Promise<string> = new Promise((resolve, reject) => {
+            if (Object.keys(toLoad).length === 0) {
+              resolve('No new packages to load');
+              return 'No new packages to load';
+            }
+      
+            const packageList = Array.from(Object.keys(toLoad));
+            _messageCallback(`Loading ${packageList.join(', ')}`)
+      
+            // monitorRunDependencies is called at the beginning and the end of each
+            // package being loaded. We know we are done when it has been called
+            // exactly "toLoad * 2" times.
+            var packageCounter = Object.keys(toLoad).length * 2;
+
+            // Add a handler for any exceptions that are thrown in the process of
+            // loading a package
+            const windowErrorHandler = (err: any) => {
+                delete self.pyodide._module.monitorRunDependencies;
+                self.removeEventListener('error', windowErrorHandler);
+                // Set up a new Promise chain, since this one failed
+                this.loadPackagePromise = new Promise((resolve) => resolve());
+                reject(err.message);
+            };
+      
+            self.pyodide._module.monitorRunDependencies = () => {
+              packageCounter--;
+              if (packageCounter === 0) {
+                for (const pkg in toLoad) {
+                  self.pyodide.loadedPackages[pkg] = toLoad[pkg];
+                }
+                delete self.pyodide._module.monitorRunDependencies;
+                self.removeEventListener('error', windowErrorHandler);
+      
+                let resolveMsg = `Loaded `;
+                if (packageList.length > 0) {
+                  resolveMsg += packageList.join(', ');
+                } else {
+                  resolveMsg += 'no packages'
+                }
+      
+                if (!IS_FIREFOX) {
+                  preloadWasm(this.pyodideModule).then(() => {
+                    console.log(resolveMsg);
+                    resolve(resolveMsg);
+                  });
+                } else {
+                  console.log(resolveMsg);
+                  resolve(resolveMsg);
+                }
+              }
+            };
+      
+            self.addEventListener('error', windowErrorHandler);
+      
+            for (const pkg in toLoad) {
+              let scriptSrc: string;
+              const packageUri = toLoad[pkg];
+              if (packageUri == 'default channel') {
+                scriptSrc = `${this.baseUrl}${pkg}.js`;
+              } else {
+                scriptSrc = `${packageUri}`;
+              }
+              // _messageCallback(`Loading ${pkg} from ${scriptSrc}`)
```

Comment:
It kind of useful to have this information, at least in the simple `console.html`, as loading large packages can take a while..

Reply:
I agree, let's turn it back on. Worst case the end-user can filter it right now by string matching (which is what I'll do for Starboard probably). In the future we can think about more structured messages (some small object).